app/controllers/application_controller.rb

```diff
@@ -53,6 +57,10 @@ def clear_saved_form(form_id)
 
   attr_reader :current_user
 
+  def set_csrf_header
+    response.set_header('X-CSRF-Token', session[:_csrf_token] || form_authenticity_token)
```

Comment:
Does rails normally re-use a session-stored token?  I was thinking Rails generates that over again for each form in normal protect mode, and was wondering if that might be a security concern?

Reply:
Yeah, you're right.  I had it generating a new one every request, but concerned it'd be too easy to end up with tokens that don't validate when they should (what if someone has two tabs open with different tokens in each?)
But, it looks like even though they are different they will decode to all be valid.  See [here](https://stackoverflow.com/questions/47723379/why-does-the-csrf-token-in-rails-not-prevent-multiple-tabs-from-working-properly/49783739#49783739).

I'm going to change it to generate a new one on each request and verify that it works in this implementation.

Reply:
Reverted.  Thanks.